lib/Analysis/Allocation.cpp

```diff
@@ -117,6 +117,9 @@ ScratchConfig getScratchConfigForCvt(RankedTensorType srcTy,
 }
 
 unsigned defaultAllocationAnalysisScratchSizeFn(Operation *op) {
+  if (op->hasAttr("allocation.size")) {
```

Comment:
I think you don't need to touch this function.

You can instead just create another `scratchSizeFn` and wrap the `defaultAllocationAnalysisScratchSizeFn`

Reply:
Yep, I think, it makes sense.

This allocation should be taken into account during all the ``ModuleAllocation`` instantiation though. Most important is common pass ``AllocateSharedMemory``.
Would it be Ok to change ``defaultAllocationAnalysisScratchSizeFn`` to new wrapper as a default parameter for ``ModuleAllocation`` constructor?